### PR TITLE
Added SiteURL test endpoint

### PR DESCRIPTION
**Pull request by threepwood-mm**
Added docs for new endpoint to test a SiteURL.

Requires https://github.com/mattermost/mattermost-server/pull/11944 to be merged

Jira ticket: https://mattermost.atlassian.net/browse/MM-14521

### CODE REVIEW

File: v4/source/system.yaml [hunk 1]
```diff
@@ -101,6 +101,53 @@
...
+                description: The Site URL to test
+      responses:
+        '200':
+          description: Site URL is valid
+          schema:
+            $ref: "#/definitions/StatusOK"
+        '400':
+          description: Site URL is not valid
+          schema:
+            $ref: "#/definitions/AppError"
```
Comment (saturninoabril) [inline v4/source/system.yaml]:
```suggestion
            $ref: '#/responses/BadRequest'
```

File: v4/source/system.yaml [hunk 2]
```diff
@@ -101,6 +101,53 @@
             // TestEmail
             ok, resp := Client.TestEmail(&config)
 
+  '/site_url/test':
+    post:
+      tags:
+        - system
+      summary: Checks the validity of a Site URL
+      description: |
+        Sends a Ping request to the mattermost server using the specified Site URL.
```
Comment (saturninoabril) [inline v4/source/system.yaml]: Please add minimum server version:
``__Minimum server version__: 5.16``

Comment (threepwood-mm) [reply v4/source/system.yaml]: Should we wait until the server PR is merged in order to figure out the version or it's for sure going to be `5.16`?

File: v4/source/system.yaml (hunk 1 above)
Comment (saturninoabril) [reply v4/source/system.yaml]: It's failing due to description and schema under '400'.  Please remove those as well, thanks

File: v4/source/system.yaml (hunk 2 above)
Comment (saturninoabril) [reply v4/source/system.yaml]: Yup, that makes sense.  Let's wait until it's merged.

**Comment by hanzei** [comment]
@saturninoabril Would you be fine merging this now as it's clearly documented that this will be in 5.16?

**Comment by saturninoabril** [comment]
Thanks for the reminder.  Done merge.

**Pull request merged**